pentaho/impl/src/main/java/pt/webdetails/cdf/dd/api/ResourcesApi.java

```diff
@@ -80,15 +80,10 @@
   public Response getResource( @QueryParam( "resource" ) @DefaultValue( "" ) String resource,
                                @HeaderParam( "if-none-match" ) String ifNoneMatch ) throws IOException {
 
-    resource = decodeAndEscape( resource );
-
     try {
-      checkExtensions( resource );
-
-      IBasicFile file = Utils.getFileViaAppropriateReadAccess( resource );
+      IBasicFile file = getResourceFile( resource );
 
       if ( file == null ) {
-        logger.error( "resource not found:" + resource );
```

Comment:
Maybe I'm missing something but is there any reason to have the logger moved into the `getResourceFile`? 
I think we could keep the verification if the file is null in one place, log the error and return the response.

Reply:
The logging is done in the new public method call, in case that some other code in the future only calls getResourceFile, they should still have that logger log an error (in the getResourceFile, it's return is a IBasicFile, so cannot have the Response builder in that). Hence the separation.

Reply:
Based on the comment on changing how we write the bytes to the output stream, the public method in resourcesApi is no longer needed, these changes will be reverted and the logging/response will be put back to its original place.